baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/safety/SafetyPropagationTransfer.java

```diff
@@ -884,24 +886,9 @@ public TransferResult<Safety, AccessPathStore<Safety>> visitLocalVariable(
         return noStoreChanges(safety, input);
     }
 
-    private static boolean isPatternBinding(LocalVariableNode node, VisitorState state) {
-        TreePath varPath = TreePath.getPath(state.getPath().getCompilationUnit(), node.getTree());
-        if (varPath == null) {
-            // Synthetic expression
-            return false;
-        }
-        TreePath parentPath = varPath.getParentPath();
-        if (parentPath == null) {
-            return false;
-        }
-        Tree enclosing = parentPath.getLeaf();
-        if (enclosing == null) {
-            return false;
-        }
-        // JCBindingPattern is newer than our compilation target, so we match strings to avoid
-        // complex build-system configurations.
-        String kindString = enclosing.getKind().name();
-        return "BINDING_PATTERN".equals(kindString);
```

Comment:
not sure why I originally checked for an enclosing binding pattern instead of the binding variable type before, most likely searching in the dark based on older jdk-11 sources